src/r1cs_nark_as/data_structures.rs

```diff
@@ -101,24 +99,6 @@ where
     }
 }
 
-/// The [`InputWitness`][input_witness] of the [`ASForR1CSNark`][as_for_r1cs_nark].
-///
-/// [input_witness]: crate::AccumulationScheme::InputWitness
-/// [as_for_r1cs_nark]: crate::r1cs_nark_as::ASForR1CSNark
-#[derive(Clone, CanonicalSerialize, CanonicalDeserialize)]
-pub struct InputWitness<F: Field> {
-    /// The sigma protocol's prover commitment of the NARK.
-    pub second_round_message: SecondRoundMessage<F>,
-}
-
-impl<F: Field> InputWitness<F> {
-    pub(crate) fn zero(witness_len: usize) -> Self {
-        Self {
-            second_round_message: SecondRoundMessage::zero(witness_len),
-        }
-    }
-}
```

Comment:
Maybe we can replace this with `pub type InputWitness<F> = SecondRoundMessage<F>`?

Reply:
That way logically the type names in the rest of this PR don't have to change from `InputWitness` to `SecondRoundMessage`, while removing the layer of indirection from before.